io.home-assistant.battery/dataset.tt

```diff
@@ -0,0 +1,178 @@
+                 "if my ${p_name} battery is low or normal",
+                 "whether my ${p_name} battery is low or normal",
+                 "if my ${p_name} battery is too low or normal",
+                 "whether my ${p_name} battery is too low or normal",
+                 "the state of the ${p_name} battery",
+                 "if the ${p_name} battery is low or normal",
+                 "whether the ${p_name} battery is low or normal",
+                 "if the ${p_name} battery is too low or normal",
+                 "whether the ${p_name} battery is too low or normal"]];
+
+  stream (p_name : String) := monitor @io.home-assistant.battery(name=p_name).state()
+  #_[utterances=["when the state of my ${p_name} battery changes",
+                 "when the state of the ${p_name} battery changes"]];
+
+  stream (p_name : String, p_state : Enum(low, normal)) := edge( @io.home-assistant.battery(name=p_name).state()) on (state == p_state)
+  #_[utterances=["when my ${p_name} battery becomes ${p_state}",
+                "when my ${p_name} battery is ${p_state}",
+                "when my ${p_name} battery turns ${p_state}",
+                "when my ${p_name} battery changes to ${p_state}",
+                "if my ${p_name} battery becomes ${p_state}",
+                "if my ${p_name} battery is ${p_state}",
+                "if my ${p_name} battery turns ${p_state}",
+                "if my ${p_name} battery changes to ${p_state}",
+                "when the ${p_name} battery becomes ${p_state}",
+                "when the ${p_name} battery is ${p_state}",
+                "when the ${p_name} battery turns ${p_state}",
+                "when the ${p_name} battery changes to ${p_state}",
+                "if the ${p_name} battery becomes ${p_state}",
+                "if the ${p_name} battery is ${p_state}",
+                "if the ${p_name} battery turns ${p_state}",
+                "if the ${p_name} battery changes to ${p_state}"]];
+
+  stream (p_name : String) := edge( @io.home-assistant.battery(name=p_name).state()) on (state == enum(low))
+  #_[utterances=["when my ${p_name} battery is low",
+                "if my ${p_name} battery is low",
+                "when the ${p_name} battery is low",
+                "if the ${p_name} battery is low",
+                "when my ${p_name} battery becomes low",
+                "if my ${p_name} battery becomes low",
+                "when the ${p_name} battery becomes low",
+                "if the ${p_name} battery becomes low",
+                "when my ${p_name} battery turns low",
+                "if my ${p_name} battery turns low",
+                "when the ${p_name} battery turns low",
+                "if the ${p_name} battery turns low",
+                "when my ${p_name} battery gets low",
+                "if my ${p_name} battery gets low",
+                "when the ${p_name} battery gets low",
+                "if the ${p_name} battery gets low"]];
```

Comment:
You don't need to copy all templates all over if all you do is replace the enum, the parametrized template above will generate the exact same sentences.